packages/react/src/components/CodeSnippet/__tests__/CodeSnippet-test.js

```diff
@@ -6,102 +6,188 @@
   });
 
-  describe('check for showMoreBtn', () => {
-    let wrapper;
+  it('should allow custom classes to be applied when passed in via className', () => {
+    render(
+      <CodeSnippet type="inline" data-testid="code-5" className="custom-class">
+        {inline}
+      </CodeSnippet>
+    );
 
-    beforeEach(() => {
-      wrapper = mount(<CodeSnippet type={'multi'} />);
-    });
+    expect(screen.getByTestId('code-5')).toHaveClass('custom-class');
+  });
 
-    it('when less then 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14',
-      });
+  it('should allow hiding the copy button', () => {
+    render(
+      <CodeSnippet type="single" hideCopyButton>
+        {single}
+      </CodeSnippet>
+    );
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(document.querySelector('button')).not.toBeInTheDocument();
+  });
 
-    it('when exactly 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15',
-      });
+  it('should set disabled on copy button if it is passed via props', () => {
+    render(
+      <CodeSnippet type="single" disabled>
+        {single}
+      </CodeSnippet>
+    );
 
-      expect(wrapper.find(Button).length).toBe(0);
-    });
+    expect(screen.getByTitle('Copy to clipboard')).toHaveAttribute('disabled');
+  });
+});
+
+describe('CodeSnippet events', () => {
+  it('should call the click handler when the copy button is clicked', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="single" onClick={onClick}>
+        {single}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTitle('Copy to clipboard');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+
+  it('should call the click handler with type inline', () => {
+    const onClick = jest.fn();
+    render(
+      <CodeSnippet type="inline" data-testid="code-6" onClick={onClick}>
+        {inline}
+      </CodeSnippet>
+    );
+
+    const button = screen.getByTestId('code-6');
+    userEvent.click(button);
+    expect(onClick).toHaveBeenCalled();
+  });
+});
+
+describe('Show more button', () => {
+  it('should not have show more button when less then 15 rows', () => {
+    render(<CodeSnippet type="multi">{multiShort}</CodeSnippet>);
+
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
+  });
+
+  it('should not have show more button when exactly 15 rows', () => {
+    render(<CodeSnippet type="multi">{multi15}</CodeSnippet>);
+
+    expect(
+      document.querySelector('button.cds--snippet-btn--expand')
+    ).not.toBeInTheDocument();
+  });
 
-    it.skip('when more then 15 rows', () => {
-      wrapper.setProps({
-        children: '1\n2\n3\n4\n5\n6\n7\n8\n9\n10\n11\n12\n13\n14\n15\n16',
-      });
+  it.skip('should have show more button when more than 15 rows', () => {
```

Comment:
```suggestion
  it('should have show more button when more than 15 rows', () => {
```

Reply:
@joshblack @tay1orjones this test still fails even with the different query selector. I'm not sure why. I've tried manually setting the number of max collapsed rows, passing the actual code as a child, making the code longer. When it logs the component, the show more button is just not there at all, even though it should be.

![Screen Shot 2022-06-22 at 5 11 19 PM](https://user-images.githubusercontent.com/32556167/175161620-a155dce0-e5fc-4036-8411-1e92f84d91e5.png)

Reply:
@jnm2377 my guess is that the layout checks that `CodeSnippet.js` is making (with like `useResizeObserver` and other logic) are using different values than what is showing up in browsers since the component is being rendered in JSDOM.

I think what this means is that `useResizeObserver` is never called so `setShouldShowMoreLess` is never set to true.

This may be something that can only be checked in the browser / using VRT for the layout to be placed as intended so dropping this makes sense to me if that is already covered